core/src/main/scala/cats/data/EnvT.scala

```diff
@@ -0,0 +1,58 @@
+package cats.data
+
+import cats.{Applicative, Comonad, Eval, Foldable, Functor, Traverse}
+
+/**
+ * EnvT[W, E, A] is the Env Comonad transformer
+ */
+final case class EnvT[W[_], E, A](wa: W[A], env: E) {
+
+  /**
+   * Return the environment
+   */
+  lazy val ask: E = env
+
+  /**
+   * Return a transformed environment
+   */
+  def asks[B](f: E => B): B = f(env)
+
+  /**
+   * Transform the environment
+   */
+  def local[E1](f: E => E1): EnvT[W, E1, A] = EnvT(wa, f(env))
+
+  def map[B](f: A => B)(implicit W: Functor[W]): EnvT[W, E, B] = EnvT(W.map(wa)(f), env)
+
+  lazy val runEnvT: (E, W[A]) = (env, wa)
```

Comment:
I'd also skip the laziness here, though you do save a tuple if it's called repeatedly.

My gut feeling is these should be returned in the same order as the constructor.  I don't have a strong feeling which order is correct.

Reply:
Makes sense to me.